test/basic-querying.test.js

```diff
@@ -958,6 +958,25 @@ describe('basic-querying', function() {
         null, // databases representing `undefined` as `null` (e.g. SQL)
       ]);
     });
+
+    describe('check __parent relationship in embedded models', () => {
+      it('should fill the parent in embedded model', async () => {
+        const [user] = await User.find({where: {name: 'John Lennon'}});
+        user.should.have.property('address').which.has.property('__parent').which.is
+          .instanceof(User).and.equals(user);
+      });
+      it('should assign the container model as parent in list property', async () => {
+        const [user] = await User.find({where: {name: 'John Lennon'}});
+        user.should.have.property('friends').which.has.property('parent').which.is
+          .instanceof(User).and.equals(user);
+      });
+      it('should have the complete chain of parents available in embedded list element', async () => {
+        const [user] = await User.find({where: {name: 'John Lennon'}});
+        user.friends.forEach((userFriend) => {
+          userFriend.should.have.property('__parent').which.equals(user.friends);
```

Comment:
Interesting. IIUC, when the model embeds a list of other models, the parent relation will be as follows:

```
Friend -> List of Friends -> User
```

I am concerned about user experience of such design, because any code accessing `__parent` must be prepared to deal with two cases (parent is a list, parent is the real parent).

Have you considered setting the `__parent` pointer directly to the owning model, skipping the intermediary list?

What are the use cases when we would need to access the List object?

Reply:
The initial need from which this feature request was created was not explicitly defined for embedded arrays. So I just chose this, since the List already contain a reference to the parent, in order to keep the chain of *containers*. I do not have a strong opinion on that. I could apply whatever you suggest

Reply:
I guess it depends on the use case you have for this new `__parent` property.

In https://github.com/strongloop/loopback-datasource-juggler/issues/1784, you wrote:

>  Right now having an embedded model instance has no relation to the parent model that contains it.

Let's say a `BlogPost` model embeds a list of `Comment` instances. When working with `Comment.__parent` property, what would you consider as the _parent model_? The list or `BlogPost`?

In my mind, List is not a model, just an implementation detail of how multiple items are stored in a single model property.

@strongloop/loopback-3x-maintainers thoughts?

Reply:
> Have you considered setting the __parent pointer directly to the owning model, skipping the intermediary list?

I agree with pointing back the the parent **directly**, which means:

> Let's say a BlogPost model embeds a list of Comment instances. When working with Comment.__parent property, what would you consider as the parent model? The list or BlogPost?

I would expect to see `BlogPost` as its parent instead of the list.

Reply:
I understand and its ok for me. Will implement this the way we discussed